### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by AbrahaKahsay**
# Project description
In this project,I have implemented the first steps of building a school library app. classes are created to represent students and teachers.

## Project requirements

- Create class Person with the following:

    - Instance vars: @id, @name, and @age.
    - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
    - Getters for @id, @name, and @age.
    - Setters for @name and @age.
    - Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
    - Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- Create class Student with the following:

    - Inherits from Person.
    - Constructor extends parent's constructor by adding @classroom and a parameter for it.
    - Method play_hooky that returns "¯\(ツ)/¯".

- Create class Teacher with the following:

    - Inherits from Person.
    - Constructor extends parent's constructor by adding @specialization and a parameter for it.
    - Override can_use_services? so it always returns true.

- Each class should be saved in a separate file.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Student < Person
+  def initialize(classroom)
+    super
+    @classroom = classroom
```
Comment (roshan-bajgain) [inline student.rb]: - [ ] According to the requirement you should add the Student class that Inherits from the Person, with the parameter `age, classroom, name = "Unknown", parent_permission = true` that is:
````ruby
def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
````

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Teacher < Person
+  def initialize(specialization)
+    super
+    @specialization = specialization
+  end
```
Comment (roshan-bajgain) [inline teacher.rb]: - [ ] According to the requirement, you should add the teacher class that Inherits from the Person, with the parameter `age, specialization, name = "Unknown", parent_permission = true` that is:
````ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @AbrahaKahsay, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 2 above)
Comment (AbrahaKahsay) [reply teacher.rb]: Thanks, @roshan-bajgain! Great catch and great lessons!

**Review by DammyShittu — APPROVED** [review]
Hi @AbrahaKahsay,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AbrahaKahsay** [comment]
@DammyShittu, Thanks!

**Pull request merged**